File: programacao/python/respostas/selecao/37.py
ano = int(input("Digite o ano: "))

d = (ano + (ano - 1) // 4 - (ano - 1) // 100 + (ano - 1) // 400) % 7

if d == 0:
    print("Domingo")
elif d == 1:
    print("Segunda-feira")
elif d == 2:
    print("Terca-feira")
elif d == 3:
    print("Quarta-feira")
elif d == 4:
    print("Quinta-feira")
elif d == 5:
    print("Sexta-feira")
else:
    print("Sabado")

File: programacao/c/respostas/selecao/37.c
#include <stdio.h>

int main(void) {
    int ano, d;

    printf("Digite o ano: ");
    scanf("%d", &ano);

    d = (ano + (ano - 1) / 4 - (ano - 1) / 100 + (ano - 1) / 400) % 7;

    if (d == 0) {
        printf("Domingo\n");
    } else if (d == 1) {
        printf("Segunda-feira\n");
    } else if (d == 2) {
        printf("Terca-feira\n");
    } else if (d == 3) {
        printf("Quarta-feira\n");
    } else if (d == 4) {
        printf("Quinta-feira\n");
    } else if (d == 5) {
        printf("Sexta-feira\n");
    } else {
        printf("Sabado\n");
    }

    return 0;
}